gempy/core/data.py

```diff
@@ -1975,6 +1975,116 @@ def create_orientation_from_surface_points(self, surface_points: SurfacePoints,
 
         return np.array([*center, *orientation, *normal])
 
+    def create_orientation_from_nn(self, surface_points,  searchcrit):
```

Comment:
I would split this function in one that gives you the nearest surface points to a given surface point.

It should look like something like `select_nearest_surfaces_points(surface_points, index, searchcrit) -> index`

Once we have the indices we can simple use `gp.set_orientation_from_surface_points` avoiding code repretition